src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -205,6 +225,62 @@ protected function isAttributeToNormalize($object, $attributeName, &$context)
     }
 
     /**
+     * Validates the submitted data and denormalizes it..
+     *
+     * @param string      $currentClass
+     * @param string      $attribute
+     * @param mixed       $data
+     * @param string|null $format
+     * @param array       $context
+     *
+     * @return mixed
+     *
+     * @throws UnexpectedValueException
+     * @throws LogicException
+     */
+    protected function validateAndDenormalize($currentClass, $attribute, $data, $format, array $context)
+    {
+        if (!$this->propertyTypeExtractor || null === $types = $this->propertyTypeExtractor->getTypes($currentClass, $attribute)){
+            return $data;
+        }
+
+        $expectedTypes = array();
+        foreach ($types as $type) {
+            if (null === $data && $type->isNullable()) {
+                return;
+            }
+
+            $builtinType = $type->getBuiltinType();
+            $class = $type->getClassName();
+            $expectedTypes[Type::BUILTIN_TYPE_OBJECT === $builtinType && $class ? $class : $builtinType] = true;
+
+            if (call_user_func('is_'.$builtinType, $data)) {
+                return $data;
+            }
+
+            if (Type::BUILTIN_TYPE_OBJECT === $builtinType) {
```

Comment:
Will this ever be executed (if I don't miss anything, the function will terminate above when `is_object()` evaluates to `true`, won't it?)?